shared/podman/images.go

```diff
@@ -6,48 +6,170 @@ package podman
+		}
 	}
 
-	if needsPull {
-		return pullImage(image, args...)
+	if strings.ToLower(pullPolicy) != "never" {
+		log.Debug().Msgf("Pulling image %s because it is missing and pull policy is not 'never'", image)
+		return image, pullImage(image, args...)
 	}
-	return nil
+
+	return image, fmt.Errorf("image %s is missing and cannot be fetched", image)
 }
 
-func checkImage(image string, pullPolicy string) (bool, error) {
-	if strings.ToLower(pullPolicy) == "always" {
-		return true, nil
+// GetRpmImageName return the RPM Image name and the tag, given an image.
+func GetRpmImageName(image string) (rpmImageFile string, tag string) {
+	rpmImageFile = strings.ReplaceAll(image, "registry.suse.com/", "")
+	rpmImageFile = strings.ReplaceAll(rpmImageFile, "/", "-")
+	parts := strings.Split(rpmImageFile, ":")
+	tag = "latest"
+	if len(parts) > 1 {
+		tag = parts[1]
+	}
+	rpmImageFile = parts[0]
+	return rpmImageFile, tag
+}
+
+// BuildRpmImagePath checks the image metadata and returns the RPM Image path.
+func BuildRpmImagePath(byteValue []byte, rpmImageFile string, tag string) (string, error) {
+	var data types.Metadata
+	if err := json.Unmarshal(byteValue, &data); err != nil {
+		return "", fmt.Errorf("cannot unmarshal: %s", err)
 	}
+	fullPathFile := rpmImageDir + data.Image.File
+	if data.Image.Name == rpmImageFile {
+		for _, metadataTag := range data.Image.Tags {
+			if metadataTag == tag {
+				return fullPathFile, nil
+			}
+		}
+	}
+	return "", nil
+}
+
+// GetRpmImagePath return the RPM image path.
+func GetRpmImagePath(image string) (string, error) {
+	log.Debug().Msgf("Looking for installed RPM package containing %s image", image)
+
+	rpmImageFile, tag := GetRpmImageName(image)
 
+	files, err := os.ReadDir(rpmImageDir)
+	if err != nil {
+		return "", fmt.Errorf("cannot read directory %s: %s", rpmImageDir, err)
+	}
+
+	for _, file := range files {
+		if !strings.HasSuffix(file.Name(), "metadata") {
+			continue
+		}
+		fullPathFileName := path.Join(rpmImageDir, file.Name())
+		log.Debug().Msgf("Parsing metadata file %s", fullPathFileName)
+		fileHandler, err := os.Open(fullPathFileName)
+		if err != nil {
+			log.Debug().Msgf("Error opening metadata file %s: %s", fullPathFileName, err)
+			continue
+		}
+		defer fileHandler.Close()
+		byteValue, err := io.ReadAll(fileHandler)
+		if err != nil {
+			log.Debug().Msgf("Error reading metadata file %s: %s", fullPathFileName, err)
+			continue
+		}
+
+		fullPathFile, err := BuildRpmImagePath(byteValue, rpmImageFile, tag)
+		if err != nil {
+			log.Warn().Msgf("Cannot unmarshal metadata file %s: %s", fullPathFileName, err)
+			return "", err
+		}
+		if len(fullPathFile) > 0 {
+			log.Debug().Msgf("%s match with %s", fullPathFileName, image)
+			return fullPathFile, nil
+		}
+		log.Debug().Msgf("%s does not match with %s", fullPathFileName, image)
+	}
+	log.Debug().Msgf("No installed RPM package containing %s image", image)
+	return "", fmt.Errorf("image %s does not exists as RPM image", image)
```

Comment:
```suggestion
	return "", nil
```
The fact that the file can't be found is not an error in itself: you don't want to break the install because of it. You are already using a check of the file path length to figure out if the RPM has been found: raising an error would have it at the same level than errors to look for the file itself.